pkg/cmd/roachtest/main.go

```diff
@@ -320,12 +320,23 @@ func updateSpecForSelectiveTests(ctx context.Context, specs []registry.TestSpec)
 	// Now, we want to take the tests common to both. These are the tests from which we need to select
 	// "successfulTestsSelectPct" percent tests to run.
 	successfulTests := make([]*testselector.TestDetails, 0)
-	for _, spec := range specs {
-		if td, ok := allTests[spec.Name]; ok && !td.Selected {
+
+	// specsMap contains a map of all the test names that are in specs
+	specsMap := make(map[string]struct{})
+	for _, test := range specs {
+		specsMap[test.Name] = struct{}{}
+	}
+
+	// allTestsMap is maintained to check for the test details while skipping a test
+	allTestsMap := make(map[string]*testselector.TestDetails)
+	for i := 0; i < len(allTests); i++ {
+		td := allTests[i]
+		if _, ok := specsMap[td.Name]; ok && !td.Selected {
 			// adding only the unselected tests that are part of the specs
 			// These are tests that have been running successfully
 			successfulTests = append(successfulTests, td)
 		}
+		allTestsMap[td.Name] = td
```

Comment:
It's okay to still set this even if the test isn't in specs right? Because it should just be ignored anyway?